frontend/src/pages/modelCatalog/screens/ModelCatalog.tsx

```diff
@@ -15,16 +15,35 @@ const ModelCatalog: React.FC = conditionalArea(
 )(() => {
   const { modelCatalogSources } = React.useContext(ModelCatalogContext);
   const renderStateProps = {
-    empty: modelCatalogSources.data.length === 0,
+    empty: modelCatalogSources.data.length === 0 || !modelCatalogSources.data[0]?.models?.length,
```

Comment:
This is kinda funky - ideally we shouldn't write anything outside of `useModelCatalogSources` that assumes we have only one source (since we'll be adding a second source soon for Neural Magic), so we don't want to do anything based on only looking at `data[0]`. I think we can assume that if we have sources they have models, so we can probably remove this line.

(Note: this is why I added the end of my comment above about `isModelCatalogSource` - if you do want to check that a source has models before trying to use it, we should do that check there instead).